model/distro/distro.go

```diff
@@ -593,3 +595,46 @@ func (d *Distro) GetResolvedPlannerSettings(s *evergreen.Settings) (PlannerSetti
 	d.PlannerSettings = resolved
 	return resolved, nil
 }
+
+func (d *Distro) Add(creator *user.DBUser) error {
+	err := d.Insert()
+	if err != nil {
+		return errors.Wrap(err, "Error inserting distro")
+	}
+	return d.AddPermissions(creator)
+}
+
+func (d *Distro) AddPermissions(creator *user.DBUser) error {
+	rm := evergreen.GetEnvironment().RoleManager()
```

Comment:
is it in anyway viable to pass the environment around rather than magic'ing it out here?